http/http-api/src/main/java/com/okta/commons/http/QueryString.java

```diff
@@ -35,18 +43,29 @@ public QueryString(){}
 
     public QueryString(Map<String,?> source) {
         super();
+
         if (!Collections.isEmpty(source)) {
-            for(Map.Entry<String,?> entry : source.entrySet()) {
+            for (Map.Entry<String,?> entry : source.entrySet()) {
                 String key = entry.getKey();
                 Object value = entry.getValue();
-                String sValue = value != null ? String.valueOf(value) : null;
+                String sValue = null;
+                if (value != null) {
+                    if (value instanceof Date) {
+                        sValue = RequestUtils.getFormattedDate(String.valueOf(value));
+                    } else {
+                        sValue = String.valueOf(value);
+                    }
+                }
                 put(key, sValue);
             }
         }
     }
 
     public String put(String key, Object value) {
         if (value != null) {
+            if (value instanceof Date) {
+                return RequestUtils.getFormattedDate(String.valueOf(value));
```

Comment:
same as above, though you may want to move all of the `toString()` logic into a new method

Reply:
done